leptos_i18n/src/fetch_locale.rs

```diff
@@ -1,43 +1,39 @@
 use crate::Locale;
 
-#[cfg(feature = "ssr")]
-#[inline]
-pub fn fetch_locale<T: Locale>() -> T {
-    crate::server::fetch_locale_server_side::<T>()
-}
-
-#[cfg(feature = "hydrate")]
-pub fn fetch_locale<T: Locale>() -> T {
-    leptos::document()
-        .document_element()
-        .and_then(|el| el.get_attribute("lang"))
-        .and_then(|lang| T::from_str(&lang))
-        .unwrap_or_default()
-}
-
-#[cfg(not(any(
-    feature = "ssr",
-    feature = "hydrate",
-    all(feature = "csr", feature = "cookie")
-)))]
-#[inline]
-pub fn fetch_locale<T: Locale>() -> T {
-    Default::default()
-}
-
-#[cfg(all(feature = "csr", feature = "cookie"))]
-pub fn fetch_locale<T: Locale>() -> T {
-    fn inner<T: Locale>() -> Option<T> {
-        let document = super::get_html_document()?;
-        let cookies = document.cookie().ok()?;
-        cookies.split(';').find_map(|cookie| {
-            let (key, value) = cookie.split_once('=')?;
-            if key.trim() == super::COOKIE_PREFERED_LANG {
-                T::from_str(value)
-            } else {
-                None
+cfg_if::cfg_if! {
```

Comment:
I'm not very familiar with `cfg_if`, but does'nt this makes the `ssr` version of the function the default when `ssr` and `hydrate`/`csr` are activated ?

Reply:
Correct, it did.

Updated to treat multiple enabled features the same as none, and follow the default path.

The server module inclusion in `lib.rs` has also been updated to match the `ssr` case, as we only use the built server retrieval functions in this branch. These feature gates will need to be kept in sync going forward.